src/components/DoctorsPage/DoctorsCard/Schedule/Schedule.tsx

```diff
@@ -32,43 +45,45 @@ const reducer = (state: IScheduleInitialState, action: ActionType): IScheduleIni
 };
 
 const Schedule: React.FC<IScheduleProps> = ({ workTimeData }) => {
-  const [stateSchedule, dispatch] = useReducer(reducer, initialState);
+  const [stateSchedule, dispatchLock] = useReducer(reducer, initialState);
+  const [disabledButton, setDisabledButton] = useState<boolean>(false);
+  const { error } = useTypesSelector((state) => state.profile);
 
   const sendWorkTime = () => {
-    dispatch({ type: ScheduleActionTypes.SUCCESS_MESSAGE });
+    dispatchLock({ type: ScheduleActionTypes.SUCCESS_MESSAGE });
   };
 
+  const viewDate = (idDateTime?: string) => {
+    dispatch(profilePatientAddAppointment(stateSchedule.choiceWorkTime._id));
+    sendWorkTime();
+  };
+  const dispatch = useDispatch();
+
   const renderItem = () =>
-    workTimeData.map((wortTimeItem, index) => {
-      if (stateSchedule.disabledItem === null) {
+    workTimeData.map((workTimeItem, index) => {
+      if (stateSchedule.disabledItem === null || index === stateSchedule.disabledItem) {
```

Comment:
do we really need to check null value? makes sense to check all falsy values

Reply:
***I did it for the following reason***
![image](https://user-images.githubusercontent.com/56519328/141970157-98967634-3f4f-483c-a089-64b838839d2a.png)

if I check for non-null then the logic will work differently

***if i have -   if (!stateSchedule.disabledItem || index === stateSchedule.disabledItem)***
![image](https://user-images.githubusercontent.com/56519328/141970380-8dfb3a91-4a2c-4728-8045-740d8ecbdf10.png)
*** and  if (stateSchedule.disabledItem || index === stateSchedule.disabledItem)***
![image](https://user-images.githubusercontent.com/56519328/141970505-b08e54ca-654a-45b4-9e70-97f8627512ec.png)

Reply:
@KLIM-nk01 I mean the condition to check only `null`, why not check for all falsy values?